### ISSUE TITLE
Can't resolve 'Fabric/fabric.require' in node_modules.

### ISSUE THREAD

**Issue by Fridthjof** [nhn/tui.image-editor#82]
Module not found: Error: Can't resolve 'fabric/dist/fabric.require' in '.\node_modules\tui-image-editor\dist'

I've already looked at former issues with fabric, these are not fixing the problem.

## Version

"tui-image-editor": "^3.2.2"
Angular 6.1.0
Node 8.11
Tried with fabric 1.6.7 and 2.4.2

```js
// Write example code
instance = new ImageEditor(document.querySelector('#tui-image-editor'),
        { cssMaxWidth: 700, cssMaxHeight: 500, selectionStyle: { cornerSize: 20, rotatingPointOffset: 70 }
        });
```

**Comment by Fridthjof** [comment]
"Fixed" the problem by replacing 
`HTTP = require("http"), HTTPS = require("https")`
into
`HTTP = require("@angular/http"), HTTPS = require("@angular/http")`

**Comment by szepeviktor** [comment]
> "postinstall": "npm install --no-save --no-optional fabric@~1.6.7"

this seems to be a work-around

**Comment by GEMI** [comment]
This issue is very annoying. How can I solve this using yarn?

**Comment by hestellez** [comment]
@Fridthjof 's solution worked for me, I also modified the **fabric.js** file.

**Comment by guilhermenunes09** [comment]
@hestellez what did you change in fabric.js?

**Comment by guilhermenunes09** [comment]
The solution here and in documentation didn't work for me, unfortunately.

**Comment by hestellez** [comment]
> 
> 
> "Fixed" the problem by replacing
> `HTTP = require("http"), HTTPS = require("https")`
> into
> `HTTP = require("@angular/http"), HTTPS = require("@angular/http")` in the fabric.require.js file.

Inside node modules, I don't know If there are problems when deploying, but it let me work for the moment

**Comment by naveenkumarmark** [comment]
I got the same issue when I tried to install it, and this command solved the issue!
` 
     npm install --save --no-optional fabric@^3.3.2
 `

**Comment by electrocnic** [comment]
There is no `fabric.require.js` in `fabric/dist/`, there is only `fabric.js` but tui-image editor is still looking for `fabric.require`

Could somebody finally please fix this? The suggested workarounds did not help in my case.

**Comment by alvarobrey** [comment]
This is because the [postinstall command in package.json](https://github.com/nhn/tui.image-editor/blob/6b23fb0acb8329954c7b546bf7bae587f3127e8a/package.json#L78) in the currently published version installs fabric version ^3.3.2, in which [there is no `fabric/dist/fabric.require` file](https://github.com/fabricjs/fabric.js/tree/6fd3bc4f66ced77e121af54f1e617c9f88b71985/dist).

Fortunately, this seems to have been patched in `master` (there is no `postinstall` or references to `fabric/dist/fabric.require`), but there is no release that contains these changes for now.
@jinwoo-kim-nhn: we really need a release for this, it's very very annoying. This would also solve incompatibilities with Yarn as per #286

**Comment by alvarobrey** [comment]
Here's a workaround:add the following to your `package.json` `scripts` section if using `yarn`:

```
    "postinstall": "yarn add --peer fabric@1.7.2 --ignore-optional --ignore-scripts"
```

Or the following if using `npm`:

```
    "postinstall": "npm install --no-optional --igore-scripts fabric@^1.7.2"
```

(The `--ignore-scripts` flag is needed so that `postinstall` does not call itself in a loop)

This is not really an ideal solution as the wrong version of `fabric` is still being installed by the `tui.image-editor` `postinstall` script, we are just then re-installing the correct version over it.

**Comment by jinwoo-kim-nhn** [comment]
We'll release the fix for # 286 next week

**Comment by jinwoo-kim-nhn** [comment]
It was released in v3.7.3.
